packages/vue-app/template/components/nuxt-link.client.js

```diff
@@ -92,15 +92,18 @@ export default {
       // Stop observing this link (in case of internet connection changes)
       observer.unobserve(this.$el)
       const Components = this.getPrefetchComponents()
+      <% if (router.linkPrefetchedClass) { %>const promises = []<% } %>
 
       for (const Component of Components) {
         const componentOrPromise = Component()
         if (componentOrPromise instanceof Promise) {
           componentOrPromise.catch(() => {})
+          <% if (router.linkPrefetchedClass) { %>promises.push(componentOrPromise)<% } %>
         }
         Component.__prefetched = true
       }<% if (router.linkPrefetchedClass) { %>
-      this.addPrefetchedClass()<% } %>
+        Promise.all(promises).then(() => this.addPrefetchedClass())
```

Comment:
We can add a return statement here to avoid unhandled promises:
```suggestion
        return Promise.all(promises).then(() => this.addPrefetchedClass())
```

Reply:
Thanks @pi0